docs/examples/plot_dynamic_beat.py

```diff
@@ -0,0 +1,132 @@
+=====================================
+
+This notebook demonstrates how to use the beat tracker in situations where the tempo
+may change over time.
+
+By default, the beat tracking algorithm [1]_ estimates a single tempo for the entire
+signal, though it does tolerate small amounts of fluctuation around that tempo.
+This is well suited for songs that have an approximately constant tempo, but where
+individual beats may not be exactly spaced accordingly.
+It is not well suited for songs that have radical shifts in tempo, eg entire sections
+that speed up or slow down, or gradual accelerations over long periods of time.
+
+The implementation in librosa (``librosa.beat.beat_track``) extends this algorithm
+to support different tempo estimates at each time point in the signal, as
+demonstrated below.
+
+.. [1] Ellis, Daniel PW. "Beat tracking by dynamic programming."
+       Journal of New Music Research 36.1 (2007): 51-60.
+       http://labrosa.ee.columbia.edu/projects/beattrack/
+"""
+
+# Code source: Brian McFee
+# License: ISC
+
+import numpy as np
+import librosa
+import matplotlib.pyplot as plt
+from IPython.display import Audio
+
+#######################################
+# ------------------------------
+# A recording with dynamic tempo
+# ------------------------------
+# The example recording included in this notebook consists of a drum beat that
+# gradually increases in tempo from 30 BPM to 240 BPM over a 30-second time interval.
+
+y, sr = librosa.load('audio/snare-accelerate.ogg')
+
+# %%
+# We can visualize the spectrogram of this recording and listen to it.
+# From the spectrogram, we can see that the spacing between drum beats becomes
+# smaller over time, indicating a gradual increase of tempo.
+
+fig, ax = plt.subplots()
+melspec = librosa.power_to_db(librosa.feature.melspectrogram(y=y, sr=sr), ref=np.max)
+librosa.display.specshow(melspec, y_axis='mel', x_axis='time', ax=ax)
+ax.set(title='Mel spectrogram')
+
+Audio(data=y, rate=sr)
+
+#########################################
+# --------------------------
+# Static tempo beat tracking
+# --------------------------
+# If we run the beat tracker on this signal directly and sonify the result as a click
+# track, we can see that it does not follow particularly well.
+# This is because the beat tracker is assuming a single (average) tempo across the
+# entire signal.
+# Note: here we are disabling trimming because we know the beats carry through to
+# the end of the signal.
+
+tempo, beats_static = librosa.beat.beat_track(y=y, sr=sr, units='time', trim=False)
+
+click_track = librosa.clicks(times=beats_static, sr=sr, click_freq=660,
+                             click_duration=0.25, length=len(y))
+
+print(f"Tempo estimate: {tempo[0]:.2f} BPM")
+Audio(data=y+click_track, rate=sr)
+
+#########################################
+# -------------
+# Dynamic tempo
+# -------------
+# Instead of using a single tempo, we can supply an array of tempo estimates
```

Comment:
Instead of estimating an aggregated tempo for the whole signal, we now estimate a time-varying tempo. For this purpose, we pass `aggregate=None` to `librosa.feature.tempo`. The return value `tempo_dynamic`, is an array: